src/Robo/Commands/Doctor/ComposerCheck.php

```diff
@@ -45,46 +57,21 @@ public function __construct(Config $config, Inspector $inspector, Executor $exec
   /**
    * Sets $this->composerJson using root composer.json file.
    *
-   * @return array
-   *   Array.
+   * @throws \Acquia\Blt\Robo\Exceptions\BltException
    */
   protected function setComposerJson() {
-    if (file_exists($this->getConfigValue('repo.root') . '/composer.json')) {
-      $composer_json = json_decode(file_get_contents($this->getConfigValue('repo.root') . '/composer.json'), TRUE);
-      $this->composerJson = $composer_json;
-
-      return $composer_json;
-    }
-
-    return [];
+    $composerJson = new ComposerJson($this->getConfigValue('repo.root'));
```

Comment:
This class never writes back to composer.json, so we don't need to store the object itself, just the file contents.